### PR TITLE
INTYGFV-16210 & INTYGFV-16211 & INTYGFV-16214: Implemented api and integration for get health care provider

### PR DESCRIPTION
**Pull request by mhornfeldt**

### CODE REVIEW

File: intyg-proxy-service/integration-fakehsa/src/main/java/se/inera/intyg/intygproxyservice/integration/fakehsa/repository/FakeHsaRepository.java [hunk 1]
```diff
@@ -67,6 +70,16 @@ public HealthCareUnit getHealthCareUnit(String id) {
     return healthCareUnitConverter.convert(parsedCareUnit);
   }
 
+  public List<HealthCareProvider> getHealthCareProvider(String id) {
+    final var parsedProvider = careProviderMap.get(id);
```
Comment (mhornfeldt) [inline intyg-proxy-service/integration-fakehsa/src/main/java/se/inera/intyg/intygproxyservice/integration/fakehsa/repository/FakeHsaRepository.java:74]: Check if we are supposed to use this map. This only has name and id so HealthCareProvider object will be rather empty

File: intyg-proxy-service/app/src/main/java/se/inera/intyg/intygproxyservice/organization/service/HealthCareProviderService.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+  private static void validateRequest(HealthCareProviderRequest request) {
+    if (request == null) {
+      throw new IllegalArgumentException("Request to get health care provider is null");
+    }
+
+    if (!isStringInvalid(request.getHsaId()) && !isStringInvalid(request.getOrganizationNumber())) {
+      throw new IllegalArgumentException(
+          String.format(
+              "HsaId '%s' and organizationNumber '%s' are both valid, only one should be valid",
```
Comment (mikaelaromild1) [inline intyg-proxy-service/app/src/main/java/se/inera/intyg/intygproxyservice/organization/service/HealthCareProviderService.java:77]: Change format to match the one in infra:
"Both hsaId and organizationNumber cannot be defined "

File: intyg-proxy-service/app/src/main/java/se/inera/intyg/intygproxyservice/organization/service/HealthCareProviderService.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+              request.getHsaId(),
+              request.getOrganizationNumber()
+          )
+      );
+    }
+
+    if (isStringInvalid(request.getHsaId()) && isStringInvalid(request.getOrganizationNumber())) {
+      throw new IllegalArgumentException(
+          String.format("HsaId '%s' and organizationNumber '%s' are both invalid",
```
Comment (mikaelaromild1) [inline intyg-proxy-service/app/src/main/java/se/inera/intyg/intygproxyservice/organization/service/HealthCareProviderService.java:86]: Change format to match the one in infra:
"One of hsaId or organizationNumber has to be defined "

**Review by mikaelaromild1 — COMMENTED** [review]
Looks good :) Added two small comments.

**Review by mikaelaromild1 — APPROVED** [review]
Well done!

**Pull request merged**